src/ray/raylet/node_manager.cc

```diff
@@ -853,11 +853,11 @@ void NodeManager::ProcessRegisterClientRequestMessage(
     DispatchTasks(local_queues_.GetReadyTasksWithResources());
   } else {
     // Register the new driver.
-    const WorkerID driver_id = from_flatbuf<WorkerID>(*message->worker_id());
+    const JobID job_id = from_flatbuf<JobID>(*message->worker_id());
```

Comment:
Hey, @pcmoritz , after this https://github.com/ray-project/ray/pull/5004 merged, the driver id is a worker id ,not a job id. so if we'd like to compute a driver_task_id from the given driver id, the current master code is fine I think.
I really don't know what went wrong with current code. I'd appreciate it very much if you can point the issue out deltailed.

Reply:
Yeah, the way it is on the current master, we have the following:
https://github.com/ray-project/ray/blob/master/python/ray/tests/test_basic.py#L2456, i.e. the job_id inside the task specification is equal to the worker_id, which is inconsistent to the definition you are suggesting (I think).

Reply:
Ok, I see. This's caused by the same value of a driver_id and its related job_id.

This PR https://github.com/ray-project/ray/pull/5110 shortens the JobID to 4 bytes, and keeps WorkerID 20 bytes. So #5110 can resolve this problem, I think. also,  
 we should distinguish driver_id from job_id in `RegisterClientRequest`. 

If you couldn't wait until #5110  merged, feel free to distinguish driver_id from job_id in your PR instead of #5099.

Reply:
Ok, that will work! I have a workaround here: https://github.com/ray-project/ray/pull/5076/files#diff-50e88810cd198c83bbce3317b1bc9924R865, we can update it once https://github.com/ray-project/ray/pull/5110 is merged :)